src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -595,6 +595,29 @@ internal bool IsEntryPointCandidate
             get { return IsStatic && Name == WellKnownMemberNames.EntryPointMethodName; }
         }
 
+        internal bool HasAsyncMainReturnType(CSharpCompilation compilation)
```

Comment:
I'll move it, but I think the name is fine as-is.  "HasAsyncMainReturnType" tells me that the return type needs to be `Task` or `Task<int>`,